resources/lib/token_decoder.py

```diff
@@ -0,0 +1,79 @@
+import re
+import math
+
+
+# the + in the beginning parses text to a number
+# a [] converts everything before into text -> therefore everythin before *10 or *100 or *1000
+# '(+((!+[]+!![]+!![]+!![]+!![]+!![]+[])+(!+[]+!![]+!![]+!![]+!![]+!![]+!![]+!![]+!![]+[])+(+!![]+[])+(!+[]+!![]+[])))>>(!+[]+!![]+!![]+!![]+!![]+!![]+!![])'
+# '(+((1+1+1+1+1+10)+(1+1+1+1+1+1+1+1+10)+(+10)+(1+10)))>>(1+1+1+1+1+1+1)'
+# 0 is a marker for times 10 or more
+# expands to
+# '(+((1+1+1+1+1+1)*1000+(1+1+1+1+1+1+1+1+1)*100+(+1)*10+(1+1)*1))>>(1+1+1+1+1+1+1)'
+
+
+class TokenDecoder:
+    #_parentheses_group_regex = r"(\(\d+\))"
+    _outer_parentheses_group_regex = r"\((?:[^)(]+|\((?:[^)(]+|\([^)(]*\))*\))*\)"
+    _lines_regex_template = r"{}\(([^\,a]+)\)"
+    TOKEN_COOKIE_NAME = 'reqkey'
+
+    def __init__(self):
+        pass
+
+    @staticmethod

```

Comment:
as the token_decoder doesnt hold state itself i havent seen the need to make it a non static class